### PR TITLE
Minor clean-ups in VMInstall and its type implementations

### PR DESCRIPTION
**Pull request by HannesWell**
## What it does

Clean-up the code in `VMInstall` and `VMInstallType` implementations in order to simplify it.

## Author checklist

- [ ] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Review by iloveeclipse — APPROVED** [review]

**Comment by HannesWell** [comment]
Thanks for your review.

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/StandardVMType.java [hunk 1]
```diff
@@ -789,21 +765,11 @@ protected LibraryInfo parseLibraryInfo(IProcess process) {
...
-			pos = index + 1;
-			index = paths.indexOf(File.pathSeparatorChar, pos);
-		}
-		String path = paths.substring(pos);
-		if (!path.equals("null")) { //$NON-NLS-1$
-			list.add(path);
-		}
-		return list.toArray(new String[list.size()]);
+		// Windows separator '\' is a RegEx meta-character, while the UNIX separator is not
+		return File.pathSeparatorChar == '/' ? paths.split(File.pathSeparator) : WINDOWS_LINE_SEPARATOR.split(paths);
```
Comment (iloveeclipse) [inline org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/StandardVMType.java:772]: This broke JVM detection. `'/'` is wrong, it must be `:`

Comment (iloveeclipse) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/StandardVMType.java:772]: Fixed via https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/437

Comment (HannesWell) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/StandardVMType.java:772]: Sorry for that, I somehow confused it with `File.separator`.
Thanks for the fix! It looks good.

Comment (iloveeclipse) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/StandardVMType.java:772]: Nothing to sorry, I've reviewed and merged that so it is my fault.

**Pull request merged**